### PR TITLE
fix(TBD-10885): refacto date and timestamp handling with dataset

### PR DESCRIPTION
**Pull request by AlixMetivier**
**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/spark/SetGlobalTimestampOptionForPreviousVersions.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                    "tMatchPredict", "tMatchModel", "tDataShuffling", "tFileInputParquet", "tHiveInput",
+                    "tDeltaLakeInput", "tDeltaLakeOutput");
+    
+    private String TIMESTAMP_OPTION = "USE_TIMESTAMP_FOR_DATASET"; //$NON-NLS-1$
+    
+    private String CHECKBOX_TIMESTAMP_NODE = "DATE_TO_TIMESTAMP_DF_TYPE_SUBSTITUTION"; //$NON-NLS-1$
+    
+    @Override
+    public Date getOrder() {
+        GregorianCalendar gc = new GregorianCalendar(2020, 01, 19, 10, 0, 0);
```
Comment (lbourgeois) [inline main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/spark/SetGlobalTimestampOptionForPreviousVersions.java:66]: Is there a special reason to be on the 2020-01-19 for this new migration task

**Review by lbourgeois — APPROVED** [review]
LGTM with a small question on the date used for the new migration task.

File: main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/spark/SetGlobalTimestampOptionForPreviousVersions.java (hunk 1 above)
Comment (AlixMetivier) [reply main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/spark/SetGlobalTimestampOptionForPreviousVersions.java:66]: absolutely not, I think I forgot to update it actually ... should it be changed ? (i don't think it overlaps with anything else)

Comment (lbourgeois) [reply main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/spark/SetGlobalTimestampOptionForPreviousVersions.java:66]: That's ok to let it like this if no overlap

**Pull request merged**